addons/addon-base-raas/packages/base-raas-services/lib/alb/alb-service.js

```diff
@@ -286,6 +286,34 @@ class ALBService extends Service {
     return result;
   }
 
+  /**
+   * Method to calculate the priority for the listener rule. The method gets the existing rules prirority
+   * and adds 1 to the maximum value
+   *
+   * @param requestContext
+   * @param resolvedVars
+   * @param listenerArn
+   * @returns {Promise<int>}
+   */
+  async calculateRulePriority(requestContext, resolvedVars, listenerArn) {
```

Comment:
Does this method need a lock since you are adding one to the priority and returning? Concurrent requests might have the same priority, is that ok?

Reply:
Added lock to the parent method calling this